### PR TITLE
APS-1410 status in placement view

### PR DESCRIPTION
**Pull request by bobmeredith**
# Context
https://dsdmoj.atlassian.net/browse/APS-1410

# Changes in this PR
Adds the placement status to the placement details page.
Also, switches to use split date/time values for actual arrival/departure dates. Once this PR is merged, the deprecated values `actualArrivalDate` and `actualDepartureDate` are no longer used by the UI and can be removed.

## Screenshots of UI changes

image

### CODE REVIEW

File: server/testutils/factories/cas1SpaceBooking.ts [hunk 1]
```diff
@@ -107,5 +129,6 @@ export default Cas1SpaceBookingFactory.define(() => {
     otherBookingsInPremisesForCrn: cas1SpaceBookingDatesFactory.buildList(4),
     deliusEventNumber: String(faker.number.int()),
     nonArrival: undefined,
-  }
+    status: 'departed',
+  } as Cas1SpaceBooking
```
Comment (froddd) [inline server/testutils/factories/cas1SpaceBooking.ts:133]: Would casting `status` above as `Cas1SpaceBookingSummaryStatus` remove the need to type cast the whole faked booking?

File: server/utils/dateUtils.ts [hunk 2]
```diff
@@ -50,6 +50,14 @@ export class DateFormats {
     return formatISO(date)
   }
 
+  /**
+   * @param date JS Date object.
+   * @returns the time in the format '19:05'.
+   */
+  static dateObjToIsoTime(date: Date) {
```
Comment (froddd) [inline server/utils/dateUtils.ts:57]: Pedantic nitpick of the day: `19:05` isn't actually ISO time -- that would be `19:05:00(.000)Z`, while `19:05` is 24hr time 😄

File: server/utils/placements/index.test.ts [hunk 3]
```diff
@@ -197,11 +197,11 @@ describe('placementUtils', () => {
           },
           {
             key: { text: 'Actual arrival date' },
-            value: { text: DateFormats.isoDateToUIDate(placement.actualArrivalDate) },
+            value: { text: DateFormats.isoDateToUIDate(placement.actualArrivalDateOnly) },
           },
           {
             key: { text: 'Arrival time' },
-            value: { text: DateFormats.timeFromDate(DateFormats.isoToDateObj(placement.actualArrivalDate)) },
+            value: { text: (placement.actualArrivalTime || '').substring(0, 5) },
```
Comment (froddd) [inline server/utils/placements/index.test.ts:204]: Is this `substring` needed here? The factory already returns time in 24hr format, I think? 🤔

**Review by froddd — APPROVED** [review]
Minor comments, seems good to go!

File: server/utils/placements/index.test.ts (hunk 3 above)
Comment (bobmeredith) [reply server/utils/placements/index.test.ts:204]: Yes, you're right. The API currently includes seconds so I originally made the factory do likewise.  Not any more though.

File: server/testutils/factories/cas1SpaceBooking.ts (hunk 1 above)
Comment (bobmeredith) [reply server/testutils/factories/cas1SpaceBooking.ts:133]: thanks, I'll try that.

**Pull request merged**